src/modules/flow/iio/nodes.c

```diff
@@ -1429,4 +1429,156 @@ light_tick(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_
     return -EIO;
 }
 
+struct proximity_data {
+    struct sol_iio_config config;
+    struct sol_drange_spec out_range;
+    double scale;
+    double offset;
+    struct sol_iio_device *device;
+    struct sol_iio_channel *channel_val;
+    bool buffer_enabled : 1;
+    bool use_device_default_scale : 1;
+    bool use_device_default_offset : 1;
+};
+
+static void
+proximity_reader_cb(void *data, struct sol_iio_device *device)
+{
+    static const char *errmsg = "Could not read channel buffer values";
+    struct sol_flow_node *node = data;
+    struct proximity_data *mdata = sol_flow_node_get_private_data(node);
+    struct sol_drange out = {
+        .min = mdata->out_range.min,
+        .max = mdata->out_range.max,
+        .step = mdata->out_range.step
+    };
+    bool b;
+
+    b = sol_iio_read_channel_value(mdata->channel_val, &out.val);
+    if (!b) goto error;
+
+    sol_flow_send_drange_value_packet(node,
+        SOL_FLOW_NODE_TYPE_IIO_LIGHT__OUT__OUT, out.val);
```

Comment:
`SOL_FLOW_NODE_TYPE_IIO_LIGHT__OUT__OUT`? Probably you wanted proximity node port...